### ISSUE TITLE
AWS credentials with SSO do not work

### ISSUE THREAD

**Issue by ChronosWS** [pulumi/pulumi#5072]
```
> pulumi version
v2.7.1

> set AWS_PROFILE=XXXX.XXXX

> aws configure list
      Name                    Value             Type    Location
      ----                    -----             ----    --------
   profile             XXXX.XXXX           manual    --profile
access_key     ****************XXXX              sso
secret_key     ****************XXXX              sso
    region                us-XXXX      config-file    ~/.aws/config
```

NOTE: The --profile above is actually coming from the AWS_PROFILE environment variable, in spite of `aws configure list` saying that it comes from a command-line argument

```
> pulumi up
...
error: unable to discover AWS AccessKeyID and/or SecretAccessKey - see https://pulumi.io/install/aws.html for details on configuration ()

> aws configure --profile XXXX.XXXX
(use values from direct IAM user rather than SSO configuration)

> aws configure list
      Name                    Value             Type    Location
      ----                    -----             ----    --------
   profile                XXXX.XXXX             manual    --profile
access_key     ****************XXXX shared-credentials-file
secret_key     ****************XXXX shared-credentials-file
    region                us-XXXX      config-file    ~/.aws/config

> pulumi up
(everything works fine)
```

NOTE: The access_key and secret_key between the two scenarios is different, since SSO generates credentials on-the-fly.  The profile name and environment variable remain unchanged.

I would expect the SSO credentials to work just like explicitly set IAM credentials.  In fact, this works for the AWS CLI.

**Comment by leezen** [comment]
This is due to the AWS Go SDK not supporting reading SSO credentials correctly: https://github.com/aws/aws-sdk-go/issues/3186

**Comment by jaxxstorm** [comment]
There are a couple ways to work around this. 
https://github.com/99designs/aws-vault will allow you to generate temporary credentials, as will this tool I wrote:

https://github.com/jaxxstorm/aws-sso-creds

**Comment by matthiassb** [comment]
@leezen This looks to be resolved in https://github.com/aws/aws-sdk-go/releases/tag/v1.37.0. Any timeline on when this might be included?

**Comment by leezen** [comment]
Apologies for not properly tracking and closing out this old issue. This was actually fixed as of v3.26.0 of the provider.
